src/node_api.cc

```diff
@@ -131,13 +131,9 @@ class BufferFinalizer : private Finalizer {
   static void FinalizeBufferCallback(char* data, void* hint) {
     std::unique_ptr<BufferFinalizer, Deleter> finalizer{
         static_cast<BufferFinalizer*>(hint)};
-    finalizer->finalize_data_ = data;
-
+    finalizer->ResetData(data);
```

Comment:
I believe in `napi_create_external_buffer`, the `BufferFinalizer` could initialize the `finalize_data` to be the actual `data`, and omit this `ResetData` call.

Reply:
I was thinking the same way, but did not want to change the existing code. Let me try to it and see if it works. It must make the code a bit simpler.

Reply:
The tests passed. I had added this change in the latest commit.